src/api.c

```diff
@@ -886,6 +889,23 @@ struct nmd_frame *nmd_get_next_frame(struct nmd_ctx *s)
         return ret_frame(s, NULL);
 
     AVFrame *frame = pop_frame(s);
+    if (!frame && s->eof) {
+        /*
+         * If the async module reached EOF and there is no more frame to
+         * return, we need to seek back to the beginning of the file because
+         * nmd_get_next_frame() is expected to loop around when reaching EOF.
+         */
+        s->eof = 0;
+
+        av_frame_free(&s->cached_frame);
+        s->last_pushed_frame_ts = AV_NOPTS_VALUE;
+
+        const struct nmdi_opts *o = &s->opts;
+        nmdi_async_seek(s->actx, o->start_time64);
+
+        return ret_frame(s, NULL);
```

Comment:
this is redundant, `frame` is `NULL` so you can rely on the common path

Reply:
Fixed.